appserver/tests/test-micro-jar/README.md

```diff
@@ -0,0 +1,25 @@
+# Payara Micro Jar Based Integration Tests
+
+This module provides plain JUnit tests that use `payara-micro.jar` as only
+compile and runtime dependency (+ junit itself).
+
+The setup makes sure the actual delivery is tested and keeps the setup simple
+and minimal. Tests do the programmtic equivalent of:
+
+        java -jar payara-micro.jar
+
+The `payara-micro.jar` is referenced using a relative path to the module that
+generates the jar. Should the jar not exist run: 
+
+        cd $PAYARA_HOME/appserver/extras/payara-micro/payara-micro-distribution
+        mvn install
+        
+Depending on the status of the build artifacts a full build might be required.
+
+Except from being dependent on a generated jar file the tests in this module
+are plain old JUnit tests that can be run using `mvn test` or in the IDE
```

Comment:
I made them into integration tests. Found an approach that is based on junit's `@Category` annotation to mark integration tests which I found a neat solution so I did that and updated the README accordingly.

Reply:
This works, but IIRC this can also be done by renaming the test classes from *Test to *IT. That should allow them to be picked up automatically by the failsafe plugin

Reply:
That triggered some memories. Seen that before. Seams like out of the box the surefire does not pick up a class named `*IT` but failsave does pick up same tests as surefire does. So this requires some explicit includes/excludes rules based on file names to work as expected. I'm not sure. I like the annotation better as it puts something in the code you can discover and go "ahh" - but I think that this is a topic the team should decide upon. @arjantijms, @Pandrex247 and @MarkWareham WDYT?

Reply:
I extracted the common test setup to a separate PR #3812 as it is used by several test PRs. Please comment there.